tests/e2e/statefulsets.go

```diff
@@ -985,6 +1046,589 @@ var _ = ginkgo.Describe("statefulset", func() {
 
 	})
 
+	/**
+	Statefulset parallel podManagementPolicy wffc
+	1. create zonal storage class
+	2. create statefull set  with replica 3 using zonal-latebinding storage class
+	3. scale down statefull set
+	4. scale up statefull set
+	5. Verify node affinity on each PV
+	6. Verify statefull pods are coming up on appropriate nodes
+	7. clean up the data
+	*/
+
+	ginkgo.It("[stretched-svc] Statefulset parallel podManagementPolicy "+
+		"wffc", ginkgo.Label(p0, vanilla, block, wcp, core), func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		ginkgo.By("Creating StorageClass for Statefulset")
+
+		// decide which test setup is available to run
+		if vanillaCluster {
+			ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+			scParameters = nil
+			storageClassName = "nginx-sc-default"
+		} else {
+			if !stretchedSVC {
+				ginkgo.By("CNS_TEST: Running for WCP setup")
+				profileID := e2eVSphere.GetSpbmPolicyID(storagePolicyName)
+				scParameters[scParamStoragePolicyID] = profileID
+				createResourceQuota(client, namespace, rqLimit, defaultNginxStorageClassName)
+				storageClassName = defaultNginxStorageClassName
+
+			}
+
+		}
+
+		if !stretchedSVC {
+			scSpec := getVSphereStorageClassSpec(storageClassName, scParameters, nil, "", "", false)
+			sc, err := client.StorageV1().StorageClasses().Create(ctx, scSpec, metav1.CreateOptions{})
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			defer func() {
+				err := client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			}()
+
+		}
+
+		ginkgo.By("Creating service")
+		service := CreateService(namespace, client)
+		defer func() {
+			deleteService(namespace, client, service)
+		}()
+
+		statefulset := GetStatefulSetFromManifest(namespace)
```

Comment:
See if you can use one of this util  for creating statefulset "createCustomisedStatefulSets" or this util "createStafeulSetAndVerifyPVAndPodNodeAffinty" 

These utils are also doing the same things as this lines of code are doing.